tests/infrastructure/vm_console_proxy/utils.py

```diff
@@ -66,3 +68,21 @@ def get_vm_console_proxy_resource(resource_kind, namespace=None):
             name=f"{TOKEN_API_VERSION}.{TOKEN_ENDPOINT}" if resource_kind == APIService else VM_CONSOLE_PROXY
         )
     return vm_console_proxy_resource_object
+
+
+def assert_resource_status_and_wait_for_availability(resource: Any, timeout: int) -> None:
+    assert resource.exists, (
+        f"Missing: {resource.kind}/{resource.name} under {getattr(resource, 'namespace', 'cluster')}"
+    )
+
+    if not hasattr(resource, "status"):
+        LOGGER.info(f"Resource {resource.kind} has no status. Skipping availability check.")
+        return
+
```

Comment:
Should not it be a bug?

Reply:
example : clusterroles  they don't have status

Reply:
In that case please don't combine validations of all in one utility. This would lead to hiding bugs. e.g. there is no check in place that ensures status check does not get skipped for non clusterrole resources.

Reply:
yeah , it could happen, so you are suggesting to check for selective resources that have status.conditions?